aws/resource_aws_media_store_container_test.go

```diff
@@ -0,0 +1,109 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/mediastore"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsMediaStoreContainer_basic(t *testing.T) {
+	rName := acctest.RandString(5)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsMediaStoreContainerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMediaStoreContainerConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMediaStoreContainerExists("aws_media_store_container.test"),
+				),
+			},
+			{
+				Config: testAccMediaStoreContainerConfig_Update(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMediaStoreContainerExists("aws_media_store_container.test"),
+					resource.TestCheckResourceAttrSet("aws_media_store_container.test", "policy"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsMediaStoreContainerDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).mediastoreconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_media_store_container" {
+			continue
+		}
+
+		input := &mediastore.DescribeContainerInput{
+			ContainerName: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.DescribeContainer(input)
+		if err != nil {
+			if ecrerr, ok := err.(awserr.Error); ok {
+				switch ecrerr.Code() {
+				case mediastore.ErrCodeContainerNotFoundException:
+					return nil
+				}
+			}
+			return err
+		}
+
+		if *resp.Container.Status != mediastore.ContainerStatusDeleting {
+			return fmt.Errorf("Media Store Container (%s) not deleted", rs.Primary.ID)
+		}
+	}
+	return nil
+}
+
+func testAccCheckAwsMediaStoreContainerExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		return nil
+	}
+}
+
+func testAccMediaStoreContainerConfig(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_media_store_container" "test" {
+  name = "tf_mediastore_%s"
+}`, rName)
+}
+
+func testAccMediaStoreContainerConfig_Update(rName string) string {
+	return fmt.Sprintf(`
+data "aws_caller_identity" "test" {}
+
+resource "aws_media_store_container" "test" {
+  name = "tf_mediastore_%s"
+	policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [{
+    "Sid": "MediaStoreFullAccess",
+    "Action": [ "mediastore:*" ],
+    "Principal": {"AWS" : "*"},
+    "Effect": "Allow",
+    "Resource": "arn:aws:mediastore:us-west-2:${data.aws_caller_identity.test.account_id}:container/tf_mediastore_%s/*",
```

Comment:
It looks like we'd benefit from having `aws_media_store_container_policy` so we can reference the container ARN instead of building it like this. 🤔 

It's certainly outside of scope of this PR, just thinking out loud as I'm reading the code...

Reply:
Yeah, I agree with you...
`"Resource"` in policy must be scope of `arn:aws:mediastore:us-west-2:1234567890:container/tf_mediastore_%s/` (I got error if I set `*` to Resource) so I thought the same when implementing.
If we'll develop `aws_media_store_container_policy`, should I remove `policy` from `aws_media_store_container`?

Reply:
No, keep it here... as I said it's outside of scope of this PR 😃

Reply:
I'm taking back what I said - see my other comment about policy.